core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -62,6 +64,10 @@
     private volatile boolean routingTableDirty = false;
 
     private volatile Future scheduledRoutingTableFuture;
```

Comment:
ok but we schedule at fixed delay so can't we just check if we are dirty due to a delayed allocation? I mean that would just be 10 sec. off in that case and we can skip the second scheduled task?